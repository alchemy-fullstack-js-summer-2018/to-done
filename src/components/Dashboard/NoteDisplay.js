import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NoteDisplay extends Component {

  static propTypes = {
    note: PropTypes.object.isRequired
  };

  render() {
    const { note } = this.props;

    return (
      <p>
        <span>{note.title}</span>: {note.notes}, {note.completed} 
      </p>
    );
  }
}

export default NoteDisplay;