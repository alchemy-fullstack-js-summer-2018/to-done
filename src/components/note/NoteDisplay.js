import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NoteDisplay extends Component {
  static propTypes = {
    note: PropTypes.object.isRequired,
    onEdit: PropTypes.func.isRequired,
    onDelete: PropTypes.func.isRequired
  };

  render() {
    const { note, onEdit, onDelete } = this.props;

    return (
      <p>
        {note.name} : {note.content}
        <button name="edit" onClick={onEdit}>Edit:</button>
        <button name="delete" onClick={onDelete}>Delete</button>
      </p>
    );
  }
}

export default NoteDisplay;