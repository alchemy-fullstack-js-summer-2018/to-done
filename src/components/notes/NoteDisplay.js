import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NoteDisplay extends Component {

  static propTypes = {
    note: PropTypes.object.isRequired,
    onEdit: PropTypes.func.isRequired,
    onDelete: PropTypes.func.isRequired
  };
  
  render() {
    const { note, onEdit, onDelete } = this.props;

    return (
      <div>
        <h3>{note.title}</h3>
        <p>{note.content}</p>
        {note.completed
          ? <p>Completed!</p>
          : <p>Not completed</p>
        }
        <button name="edit" onClick={onEdit}>Edit</button>
        <button name="delete" onClick={onDelete}>delete</button>
      </div>
    );
  }
}
 
export default NoteDisplay;