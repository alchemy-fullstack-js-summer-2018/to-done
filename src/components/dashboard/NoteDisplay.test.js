import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import NoteDisplay from './NoteDisplay';

describe('NoteDisplay', () => {

  it('renders display', () => {
    const onEdit = jest.fn();
    const promise = Promise.resolve();
    onEdit.mockReturnValueOnce(promise);

    const onDelete = jest.fn();
    onDelete.mockReturnValueOnce(promise);

    const note = { 
      key: '123', 
      title: 'note', 
      content: 'some content', 
      completed: false 
    };

    const wrapper = shallow(<NoteDisplay
      note={note}
      onEdit={onEdit}
      onDelete={onDelete}
    />);

    expect(toJSON(wrapper)).toMatchSnapshot();

    return promise
      .then(() => {
        wrapper.update();
        expect(toJSON(wrapper)).toMatchSnapshot();
      });
  });

});
