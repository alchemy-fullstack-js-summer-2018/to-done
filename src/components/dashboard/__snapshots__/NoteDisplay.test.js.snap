// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteDisplay renders display 1`] = `
<div>
  <h3>
    note
  </h3>
  <p>
    some content
  </p>
  <button
    name="edit"
    onClick={[MockFunction]}
  >
    ✏️
  </button>
  <button
    name="delete"
    onClick={[MockFunction]}
  >
    🗑
  </button>
</div>
`;

exports[`NoteDisplay renders display 2`] = `
<div>
  <h3>
    note
  </h3>
  <p>
    some content
  </p>
  <button
    name="edit"
    onClick={[MockFunction]}
  >
    ✏️
  </button>
  <button
    name="delete"
    onClick={[MockFunction]}
  >
    🗑
  </button>
</div>
`;
